ray_on_golem/server/services/golem/manager_stack.py

```diff
@@ -36,3 +72,119 @@ async def stop(self) -> None:
                     logger.exception(f"{manager} stop failed!")
 
         logger.info("Stopping stack managers done")
+
+    async def get_agreement(self) -> Agreement:
+        return await self._agreement_manager.get_agreement()
+
+    @classmethod
+    async def create(
+        cls,
+        node_config: NodeConfigData,
+        payment_network: str,
+        is_head_node: bool,
+        payment_manager: PaymentManager,
+        demand_config_helper: DemandConfigHelper,
+        golem: GolemNode,
+    ) -> "ManagerStack":
+        stack = cls()
+        extra_proposal_plugins: Dict[str, ProposalManagerPlugin] = {}
+        extra_proposal_scorers: Dict[str, ProposalScorer] = {}
+
+        payloads = await demand_config_helper.get_payloads_from_demand_config(node_config.demand)
+        demand_lifetime = DEFAULT_DEMAND_LIFETIME
+
+        ManagerStackNodeConfigHelper.apply_budget_control_expected_usage(
+            extra_proposal_plugins, extra_proposal_scorers, node_config
+        )
+        ManagerStackNodeConfigHelper.apply_budget_control_hard_limits(
+            extra_proposal_plugins, node_config
+        )
+        ManagerStackNodeConfigHelper.apply_priority_head_node_scoring(
+            extra_proposal_scorers, node_config
+        )
+
+        proposal_negotiators = [PaymentPlatformNegotiator()]
+        if node_config.budget_control.payment_interval_hours is not None:
+            logger.debug(
+                "Adding mid agreement payments based on given payment_interval: %s",
+                node_config.budget_control.payment_interval_hours,
+            )
+
+            minimal_payment_timeout = timedelta(
+                hours=node_config.budget_control.payment_interval_hours.minimal
+            )
+            optimal_payment_timeout = timedelta(
+                hours=node_config.budget_control.payment_interval_hours.optimal
+            )
+
+            payloads.append(
+                PaymentInfo(
+                    debit_notes_accept_timeout=int(
+                        DEFAULT_DEBIT_NOTES_ACCEPT_TIMEOUT.total_seconds()
+                    ),
+                    debit_notes_interval=int(DEFAULT_DEBIT_NOTE_INTERVAL.total_seconds()),
+                    payment_timeout=int(minimal_payment_timeout.total_seconds()),
+                )
+            )
+            demand_lifetime = DEFAULT_LONG_RUNNING_DEMAND_LIFETIME
+
+            proposal_negotiators.append(
+                MidAgreementPaymentsNegotiator(
+                    min_debit_note_interval=DEFAULT_DEBIT_NOTE_INTERVAL,
+                    optimal_debit_note_interval=DEFAULT_DEBIT_NOTE_INTERVAL,
+                    min_payment_timeout=minimal_payment_timeout,
+                    optimal_payment_timeout=optimal_payment_timeout,
+                )
+            )
+
+        demand_manager = ManagerStackNodeConfigHelper.prepare_demand_manager_for_node_type(
```

Comment:
Woah, that's a regression since #188! Good catch! Also, it looks like managers with start/stop need some kind of warning if they are used while not running.

Reply:
Whoops made a change too quickly and did not see that coming! Demand manager is already added to the stack... but inside of the helper, due to conditionally having multiple managers inside.